lib/language_pack/ruby.rb

```diff
@@ -1018,6 +1018,26 @@ def add_node_js_binary
     if Pathname(build_path).join("package.json").exist? ||
          bundler.has_gem?('execjs') ||
          bundler.has_gem?('webpacker')
+
+      version = @node_installer.version
+      old_version = @metadata.fetch("default_node_version") { version }
+
+      if version != version
+        warn(<<~WARNING, inline: true)
+          Default version of Node.js changed (#{old_version} to #{version})
+        WARNING
+      end
+
+      warn(<<~WARNING, inline: true)
+        Installing a default version (#{version}) of Node.js.
+        This version is not pinned and can change over time, causing unexpected failures.
+
+        Heroku recommends placing the `heroku/nodejs` buildpack in front of
+        `heroku/ruby` to install a specific version of node:
```

Comment:
Do we need to mention "(and define `engines.node` in `package.json`)" here too? Otherwise adding `heroku/nodejs` will still install a default version, just a different default (albeit using `heroku/nodejs` is still recommended, even if relying on default versions).

Reply:
I'm linking to the docs that mention it. I would prefer that the `heroku/nodejs` buildpack has a very strong recommendation and complete documentation in the output instead of duplicating it both here and in the Ruby docs. I'll be willing to reconsider with new information (or simply again at a later date).

Reply:
I agree the focus should be on "we recommend everyone use `heroku/nodejs`" rather than the default version part. The issue is that the wording here (a) implies that the only reason to use `heroku/nodejs` is if wanting to customise the version, (b) then gives slightly misleading instructions on how to customise the version (which could lead to confusion if the user doesn't follow the link).

What do you think about adjusting the wording to say roughly "we recommend you use the `heroku/nodejs` buildpack in front of the `heroku/ruby` buildpack as it offers more comprehensive Node.js support, including the ability to customise the Node.js version"?

Reply:
Updated in https://github.com/heroku/heroku-buildpack-ruby/pull/1402